pkg/stub/handler.go

```diff
@@ -2,96 +2,410 @@ package stub
+	templateclientwrapper TemplateClientWrapper
+	secreteclientwrapper  SecretClientWrapper
+
+	fileimagegetter    ImageStreamFromFileGetter
+	filetemplategetter TemplateFromFileGetter
+	filefinder         ResourceFileLister
+
+	namespace string
+
+	skippedTemplates    map[string]bool
+	skippedImagestreams map[string]bool
+}
+
+func (h *Handler) manageDockerCfgSecret(deleted bool) error {
+	err := h.secreteclientwrapper.Delete("openshift", h.registrySecret.Name, &metav1.DeleteOptions{})
+	if err != nil && !kerrors.IsNotFound(err) {
+		return h.processError(v1alpha1.SecretUpdateFailed, err, "failed to delete before create dockerconfig secret the openshift namespace: %v")
+	}
+	if deleted {
+		logrus.Printf("registry dockerconfig secret %s was deleted", h.registrySecret.Name)
+		h.registrySecret = nil
+		h.operatorCfg.Status.SecretCreated = false
+		h.sdkwrapper.Update()
+	} else {
+		secretToCreate := corev1.Secret{}
+		h.registrySecret.DeepCopyInto(&secretToCreate)
+		secretToCreate.Namespace = ""
+		secretToCreate.ResourceVersion = ""
+
+		_, err = h.secreteclientwrapper.Create("openshift", &secretToCreate)
+		if err != nil {
+			return h.processError(v1alpha1.SecretUpdateFailed, err, "failed to create registry dockerconfig secret in openshift namespace : %v")
+		}
+		logrus.Printf("created/updated dockerconfig secret %s in openshift namespace", h.operatorCfg.Spec.RegistryCredentials.Name)
+
+		h.operatorCfg.Status.SecretCreated = true
+		h.sdkwrapper.Update()
+
+	}
+
+	status := v1alpha1.ResourceCondition{
+		Type: v1alpha1.SecretUpdateComplete,
+		Time: kapis.Now(),
+	}
+	h.operatorCfg.Status.Conditions = append(h.operatorCfg.Status.Conditions, status)
+	err = h.sdkwrapper.Update()
+	if err != nil {
+		if !kerrors.IsConflict(err) {
+			return h.processError(v1alpha1.SamplesUpdateFailed, err, "failed adding success condition to config: %v")
+		}
+		logrus.Println("got conflict error on success status update, just going to forgo this time vs. retry")
+	}
+
+	return nil
 }
 
 func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
-	switch o := event.Object.(type) {
+	switch event.Object.(type) {
+	case *corev1.Secret:
+		dockercfgSecret, _ := event.Object.(*corev1.Secret)
+
+		if h.operatorCfg == nil {
+			h.registrySecret = dockercfgSecret
+			return nil
+		}
+
+		if h.operatorCfg.Spec.RegistryCredentials.Name == dockercfgSecret.Name {
+			if h.registrySecret != nil &&
+				h.registrySecret.ResourceVersion == dockercfgSecret.ResourceVersion &&
+				h.operatorCfg.Status.SecretCreated &&
+				!event.Deleted {
+				return nil
+			}
+
+			h.registrySecret = dockercfgSecret
+
+			err := h.manageDockerCfgSecret(event.Deleted)
+			if err != nil {
+				return err
+			}
+		}
+
 	case *v1alpha1.SamplesResource:
-		logrus.Printf("ctrl cfg %#v", o)
-		restconfig, err := getRestConfig()
-		if err != nil {
-			logrus.Errorf("failed to get rest config : %v", err)
-			return err
+		if event.Deleted {
+			h.operatorCfg = nil
```

Comment:
what actually happens in this scenario?  I think we should be recreating the default SamplesResource

Reply:
Deleting the SamplesResource object would allow a human to turn off the operator.

You'd have to clarify exactly how we recreate the default operator before I could weigh in on the value of it.  Just the version cached in memory?  Recreate the empty object in etcd and assume our architecture and distribution (centos/rhel) defaults?

Reply:
Turning off/pausing the operator will be handled differently.  (Still being discussed exactly what the mechanism will be).  

But the general expectation is that every operator, when it starts up and does not see a relevant operator-resource instance (in this case samplesresource), will create a default one, and then get an event for that operator-resource being created, and handle it by instantiating whatever it's supposed to operate. 

So for the samples operator the auto-generated samplesresource would have to contain appropriate arch+distribution config, which we'd need to get from the "global cluster config" we've discussed.  For now you can hardcode it has x86+centos though :)

This is "4a" in the "startup" workflow of the samples operator proposal.  But since we can be restarted at anytime, our reaction to seeing the resource deleted should be the same as if we came up and it was missing.  So either we should do nothing at all (pending a restart), or we should do the same logic we would do on startup.